yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+        renderedRepresentation
+      }
+      duration
+      promotionalHeadline
+      promotionalMedia {
+        ... on Image {
+          crops {
+            name
+            renditions {
+              name
+              width
+              height
+              url
+            }
+          }
+        }
+      }
+      renditions {
+        type
+        width
+        height
+        url
+        bitrate
+      }
+      summary
+    }
+  }
+}'''
+
+    def _build_playlist(self, media_items):
+        entries = []
+        for media_id in media_items:
+            json_obj = traverse_obj(self._json_from_graphql(media_id) or {}, ('data', 'video'))
+
+            title = json_obj.get('promotionalHeadline') or media_id
+            description = json_obj.get('summary')
+            creators = ', '.join(
+                [remove_start(creator, 'By ') for creator in traverse_obj(
+                    json_obj, ('bylines', ..., 'renderedRepresentation'))])
+            duration = int_or_none(json_obj.get('duration'))
+            media_content = json_obj.get('renditions')
+            formats, subtitles = self._extract_media_from_json(media_id, media_content)
+
+            thumbnails = []
+            for image in traverse_obj(json_obj, ('promotionalMedia', 'crops', ..., 'renditions', ...)):
+                image_url = image.get('url')
+                if not url_or_none(image_url):
+                    continue
+                thumbnails.append({
+                    'name': image.get('name'),
+                    'url': image_url,
+                    'width': int_or_none(image.get('width')),
+                    'height': int_or_none(image.get('height')),
+                })
+
+            entries.append({
+                'id': media_id,
+                'title': title,
+                'description': description,
+                'duration': duration,
+                'creator': creators,
+                'formats': formats,
+                'subtitles': subtitles,
+                'thumbnails': thumbnails,
+            })
+        return entries
+
+    def _json_from_graphql(self, media_id):
+        # reference: `id-to-uri.js`
+        namespace = uuid.UUID(self._DNS_UUID)
+        video_uuid = uuid.uuid5(namespace, 'video')
+        media_uuid = uuid.uuid5(video_uuid, media_id)
+
+        payload = {
+            'query': self._GRAPHQL_QUERY,
+            'variables': {'id': f'nyt://video/{media_uuid}'}
+        }
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Nyt-App-Type': 'vhs',
+            'Nyt-App-Version': 'v3.52.21',
+            'Nyt-Token': self._TOKEN,
+            'Origin': 'https://cooking.nytimes.com',
+            'Referer': 'https://www.google.com/',
+        }
+
+        return self._download_json(
+            self._GRAPHQL_API, id, note='Downloading json from GRAPHQL API',
+            data=json.dumps(payload, separators=(',', ':')).encode(), headers=headers, fatal=False)
```

Comment:
if this always non-fatal, we could default to `{}` in the return instead after the call
```suggestion
            data=json.dumps(payload, separators=(',', ':')).encode(), headers=headers, fatal=False) or {}
```